Assets/MixedRealityToolkit.SDK/Features/Utilities/PointerUtils.cs

```diff
@@ -144,6 +144,159 @@ public static IEnumerable<IMixedRealityPointer> GetPointers()
+        /// <returns>
+        /// True if the ray contains valid data, false otherwise.
+        /// </returns>
+        public static bool TryGetHandRay(Handedness hand, out Ray ray)
+        {
+            return TryGetRay(InputSourceType.Hand, hand, out ray);
+        }
+
+        /// <summary>
+        /// Gets the physics ray based on the interaction enabled pointer assosiciated with the
+        /// motion controller.
+        /// </summary>
+        /// <param name="hand">The handedness of the motion controller</param>
+        /// <param name="ray">The physics ray for the pointer</param>
+        /// <returns>
+        /// True if the ray contains valid data, false otherwise.
+        /// </returns>
+        public static bool TryGetMotionControllerRay(Handedness hand, out Ray ray)
+
+        {
+            return TryGetRay(InputSourceType.Controller, hand, out ray);
+        }
+
+        /// <summary>
+        /// Gets the physics ray based on the interaction enabled pointer assosiciated with the user's hand
+        /// or a motion controller.
+        /// </summary>
+        /// <param name="hand">The handedness of the user's hand or motion controller</param>
+        /// <param name="ray">The physics ray for the pointer</param>
+        /// <param name="sourceType">The type of input source</param>
+        /// <returns>
+        /// True if the ray contains valid data, false otherwise.
+        /// </returns>
+        public static bool TryGetHandOrControllerRay(Handedness hand, out Ray ray, out InputSourceType sourceType)
+        {
+            ray = new Ray();
+            sourceType = InputSourceType.Other;
+
+            // First, try to get the hand ray
+            if (TryGetHandRay(hand, out ray))
+            {
+                sourceType = InputSourceType.Hand;
+                return true;
+            }
+
+            // Then try to get the controller ray
+            if (TryGetMotionControllerRay(hand, out ray))
+            {
+                sourceType = InputSourceType.Controller;
+                return true;
+            }
+
+            return false;
+        }
+
+        /// <summary>
+        /// Gets the physics ray based on the interaction enabled pointer assosiciated with the
+        /// desired input source type.
+        /// </summary>
+        /// <param name="sourceType">The type of input source</param>
+        /// <param name="hand">The handedness of the input source</param>
+        /// <param name="ray">The physics ray for the pointer</param>
+        /// <returns>
+        /// True if the ray contains valid data, false otherwise.
+        /// </returns>
+        public static bool TryGetRay(InputSourceType sourceType, Handedness hand, out Ray ray)
+        {
+            ray = new Ray();
+
+            foreach (IMixedRealityPointer pointer in PointerUtils.GetPointers<IMixedRealityPointer>(hand, sourceType))
+            {
+                if (pointer.IsInteractionEnabled)
```

Comment:
If my left hand is up / detected by the system, but it is near a grabbable, then the line pointer will actually not be active / interaction enabled will be false. This means that this method could return false even if a hand is detected by the system. And actually reading the method even further, it could return false sometimes (say, if the system disabled grab pointers and line pointers), return the actual hand pointing ray sometimes (say, if the hand is not near a grabbable or touchable, and line pointers are enabled), and return the position and direction of the grab pointer sometimes (if the hand is near a grabbable and grab, line pointers are enabled). 

This would probably be very confusing for developers, since I assume what developers expect when they call TryGetRay(InputSourceType.hand, Handedness.Right) is to get the position, direction of the hand ray -- the line that shoots out when the hands are not near anything.

Reply:
fixed